### ISSUE TITLE
YOLO produces exact same results when trained with different seeds

### ISSUE THREAD

**Issue by emcrobert** [ultralytics/ultralytics#6820]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Train

### Bug

I first noticed this when running experiments to help select hyperparameters. Ran experiments multiple times with different seeds. I expected the different seeds would result in different initial weights being chosen, data being shuffled differently etc and result in models with slightly different performance for each seed. 

When using RT-DETR model each different seed resulted in slightly different train, validation and test metrics, whilst YOLO produced the exact same performance no matter the seed.

For example the validation performance of Yolo-l trained on COCO for 10 epochs with seed 0,1 and 2 is below:
Box(P          R      mAP50  mAP50-95)
0.898      0.836      0.901      0.751
0.898      0.836      0.901      0.751
0.898      0.836      0.901      0.751

Whilst doing the same for RT-DETR produces different metrics for each seed (as you would expect)
Box(P          R      mAP50  mAP50-95)
 0.874      0.769      0.877       0.69
0.814      0.805      0.879      0.691
0.854      0.796      0.878      0.694

I'm aware I haven't set the deterministic which defaults to true. However, whilst I would expect deterministic being set to true to result in the same performance each time for a particular seed, I would still expect results to be dependent upon the seed.

### Environment

Ultralytics YOLOv8.0.167 🚀 Python-3.10.12 torch-2.0.1+cu117 CUDA:0 (NVIDIA GeForce RTX 3090, 24260MiB)
Setup complete ✅ (32 CPUs, 251.5 GB RAM, 629.2/863.2 GB disk)

### Minimal Reproducible Example

```
from ultralytics import YOLO
from ultralytics import RTDETR
import torch


for cur_model_name in ['yolov8l.pt', 'rtdetr-l.pt']:
    for run_no in [1,2,3]:

        # load model
        if "yolo" in cur_model_name:
            model = YOLO(cur_model_name)
        elif "detr" in cur_model_name:
            model = RTDETR(cur_model_name)


        results = model.train(data='coco128.yaml',
                              seed=run_no,
                              epochs=10,
                              device=torch.cuda.current_device(),
                              )
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@emcrobert thanks for the bug report. Yes, changing the seed would make one naturally expect to see a different result. I'll add a TODO to investigate!

**Comment by glenn-jocher** [comment]
@emcrobert 👋 hi, thanks for letting us know about this possible problem with YOLO 🚀. We've created a few short guidelines below to help users provide what we need in order to start investigating a possible problem. When I test with different seeds I get different results using this command. When I repeat a training with the same seed, I get a deterministic result, so everything for me is running correctly.

```
(venv) glennjocher@Glenns-MacBook-Air-2 ultralytics % yolo train imgsz=320 epochs=10 data=coco8.yaml seed=1236
```

## How to create a Minimal, Reproducible Example

When asking a question, people will be better able to provide help if you provide **code** that they can easily understand and use to **reproduce** the problem. This is referred to by community members as creating a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). Your code that reproduces the problem should be:

- ✅ **Minimal** – Use as little code as possible to produce the problem
- ✅ **Complete** – Provide all parts someone else needs to reproduce the problem
- ✅ **Reproducible** – Test the code you're about to provide to make sure it reproduces the problem

For [Ultralytics](https://ultralytics.com/) to provide assistance your code should also be:

- ✅ **Current** – Verify that your code is up-to-date with GitHub with `git pull` or `pip install -U ultralytics` to ensure your problem has not already been solved in the latest code version.
- ✅ **Unmodified** – Your problem must be reproducible using official Ultrlaytics code without changes. [Ultralytics](https://ultralytics.com/) does not provide support for custom code ⚠️.

If you believe your problem meets all the above criteria, please close this issue and raise a new one using the 🐛 **Bug Report** [template](https://github.com/ultralytics/ultralytics/issues/new/choose) with a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us better understand and diagnose your problem. 

Thank you! 😃

**Comment by ILIAHHne63** [comment]
The same problem, I try to run YoLO 2 times, but metrics are same.

model = YOLO('yolo10n.yaml').load('my_model_yolo.pt')
 model.train(
        project=project, 
        data=data_path, 
        epochs=EPOCHS,
        plots=True, 
        seed=1236,
        deterministic=False,
        save_period=SAVE_PERIOD
    )

**Comment by pderrenger** [comment]
Thank you for your comment. Please ensure you are using the latest version of the Ultralytics package by running `pip install -U ultralytics`. If the issue persists, try setting `deterministic=True` to see if it affects the reproducibility. If you continue to experience the same metrics, please provide more details about your environment and any additional configurations.
